registry/zookeeper/registry.go

```diff
@@ -290,20 +290,17 @@ func (r *zkRegistry) getCloseListener(conf *common.URL) (*RegistryConfigurationL
 
 	var zkListener *RegistryConfigurationListener
 	r.dataListener.mutex.Lock()
+	defer r.dataListener.mutex.Unlock()
```

Comment:
If this lock use defer, will include [listenerLock](https://github.com/apache/dubbo-go/blob/234633932aab82913ede37c3f32c7a48a912b869/registry/zookeeper/registry.go#L309), is it right logic?

Reply:
No.....getListener use  defer too, so here may be cause dead lock.